pkg/reconciler/utils/filter.go

```diff
@@ -0,0 +1,26 @@
+/*
+Copyright 2020 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package utils
+
+import (
+	brokerv1beta1 "github.com/google/knative-gcp/pkg/apis/broker/v1beta1"
+	eventingv1beta1 "knative.dev/eventing/pkg/apis/eventing/v1beta1"
+	reconciler "knative.dev/pkg/reconciler"
+)
+
+// BrokerClassFilter is the function to filter brokers with proper brokerclass.
+var BrokerClassFilter = reconciler.AnnotationFilterFunc(eventingv1beta1.BrokerClassAnnotationKey, brokerv1beta1.BrokerClass, false /*allowUnset*/)
```

Comment:
nits: maybe we can make it `BrokerClassAnnotationFilter`?

Reply:
hmm, I personally prefer `BrokerClassFilter` because the util function name should expose what's filtered (the broker class) not how it's filtered (through an annotation). I feel adding `Annotation` into the name is exposing implementation details the caller usually don't care.

Reply:
Thanks, I see, that makes sense. I saw the comment in https://github.com/google/knative-gcp/pull/2099/files#diff-88052577c3d76c3adb4cda9795e0020e8a949de34b610b34dadace01fab839c4R105 so I thought the implementation detail is already leaked. Maybe we can change that comment.

Reply:
Thanks a lot for updating the comments!